test/js-api/gc/casts.tentative.any.js

```diff
@@ -0,0 +1,303 @@
+  assert_throws_js(TypeError, () => exports.concreteStructArg(42));
+  assert_throws_js(TypeError, () => exports.concreteStructArg(42n));
+  assert_throws_js(TypeError, () => exports.concreteStructArg("foo"));
+  assert_throws_js(TypeError, () => exports.concreteStructArg({}));
+  assert_throws_js(TypeError, () => exports.concreteStructArg(exports.testFunc));
+  assert_throws_js(TypeError, () => exports.concreteStructArg(() => {}));
+  assert_throws_js(TypeError, () => exports.concreteStructArg(null));
+
+  exports.concreteStructNullableArg(null);
+  exports.concreteStructNullableArg(exports.makeStruct());
+  assert_throws_js(TypeError, () => exports.concreteStructNullableArg(exports.makeStruct2()));
+  assert_throws_js(TypeError, () => exports.concreteStructNullableArg(exports.makeArray()));
+  assert_throws_js(TypeError, () => exports.concreteStructNullableArg(42));
+  assert_throws_js(TypeError, () => exports.concreteStructNullableArg(42n));
+  assert_throws_js(TypeError, () => exports.concreteStructNullableArg("foo"));
+  assert_throws_js(TypeError, () => exports.concreteStructNullableArg({}));
+  assert_throws_js(TypeError, () => exports.concreteStructNullableArg(exports.testFunc));
+  assert_throws_js(TypeError, () => exports.concreteStructNullableArg(() => {}));
+}, "concrete struct casts");
+
+test(() => {
+  exports.concreteArrayArg(exports.makeArray());
+  assert_throws_js(TypeError, () => exports.concreteArrayArg(exports.makeArray2()));
+  assert_throws_js(TypeError, () => exports.concreteArrayArg(exports.makeStruct()));
+  assert_throws_js(TypeError, () => exports.concreteArrayArg(42));
+  assert_throws_js(TypeError, () => exports.concreteArrayArg(42n));
+  assert_throws_js(TypeError, () => exports.concreteArrayArg("foo"));
+  assert_throws_js(TypeError, () => exports.concreteArrayArg({}));
+  assert_throws_js(TypeError, () => exports.concreteArrayArg(exports.testFunc));
+  assert_throws_js(TypeError, () => exports.concreteArrayArg(() => {}));
+  assert_throws_js(TypeError, () => exports.concreteArrayArg(null));
+
+  exports.concreteArrayNullableArg(null);
+  exports.concreteArrayNullableArg(exports.makeArray());
+  assert_throws_js(TypeError, () => exports.concreteArrayNullableArg(exports.makeArray2()));
+  assert_throws_js(TypeError, () => exports.concreteArrayNullableArg(exports.makeStruct()));
+  assert_throws_js(TypeError, () => exports.concreteArrayNullableArg(42));
+  assert_throws_js(TypeError, () => exports.concreteArrayNullableArg(42n));
+  assert_throws_js(TypeError, () => exports.concreteArrayNullableArg("foo"));
+  assert_throws_js(TypeError, () => exports.concreteArrayNullableArg({}));
+  assert_throws_js(TypeError, () => exports.concreteArrayNullableArg(exports.testFunc));
+  assert_throws_js(TypeError, () => exports.concreteArrayNullableArg(() => {}));
+}, "concrete array casts");
+
+test(() => {
+  exports.concreteFuncArg(exports.testFunc);
```

Comment:
Maybe have another one with a testFunc2 here and below, analogous to structs and arrays.